### PR TITLE
Timescale fix

### PR DESCRIPTION
**Pull request by PavelVanecek**
## Description

Fixes regression https://www.chromatic.com/test?appId=63da8268a0da9970db6992aa&id=66aa465be316cf9b344d1a0b introduced in https://github.com/recharts/recharts/pull/4840

## Related Issue

https://github.com/recharts/recharts/issues/4583

### CODE REVIEW

File: src/state/selectors/axisSelectors.ts [hunk 1]
```diff
@@ -870,12 +870,20 @@ function getD3ScaleFromType(realScaleType: string | undefined) {
...
   realScaleType: string | undefined,
   axisDomain: NumberDomain | CategoricalDomain,
   axisRange: [number, number],
 ): RechartsScale | undefined {
+  if (axisDomain == null) {
+    return undefined;
+  }
+  if (typeof axis.scale === 'function') {
+    // @ts-expect-error we're going to assume here that if axis.scale is a function then it is a d3Scale function
+    return axis.scale.copy().domain(axisDomain).range(axisRange);
```
Comment (ckifer) [inline src/state/selectors/axisSelectors.ts:883]: I guess we'll find out when we release if this is an issue 😅

**Review by ckifer — APPROVED** [review]

File: src/state/selectors/axisSelectors.ts (hunk 1 above)
Comment (PavelVanecek) [reply src/state/selectors/axisSelectors.ts:883]: This has always been the case so it shouldn't be an issue.

Comment (PavelVanecek) [reply src/state/selectors/axisSelectors.ts:883]: Except before my changes this assumption was hidden behind an ocean of `any`s 😬

Comment (ckifer) [reply src/state/selectors/axisSelectors.ts:883]: Of course it was haha

**Pull request merged**